### PR TITLE
feat: Add oauth integration for partner applications

### PR DESCRIPTION
**Pull request by sfc-gh-jcieslak**
Changes:
- New oauth integration for partner applications added
- Examples added
- Migration Guide was added in the previous pr (oauth for custom clients)
- Common description for not detecting external changes and using manual re-creation added
- Left TODOs for UNSETs not working
- Address things from https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2868

## Test Plan

* [x] acceptance tests

## References

* [CREATE](https://docs.snowflake.com/en/sql-reference/sql/create-security-integration-oauth-snowflake)

### CODE REVIEW

**Review by sfc-gh-jmichalak — APPROVED** [review]

File: pkg/datasources/security_integrations_acceptance_test.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func TestAcc_SecurityIntegrations_Scim(t *testing.T) {
 					resource.TestCheckResourceAttrSet("data.snowflake_security_integrations.test", "security_integrations.0.show_output.0.created_on"),
 
 					resource.TestCheckResourceAttr("data.snowflake_security_integrations.test", "security_integrations.0.describe_output.#", "1"),
-					resource.TestCheckResourceAttr("data.snowflake_security_integrations.test", "security_integrations.0.describe_output.0.todo.#", "1"),
-					resource.TestCheckResourceAttrSet("data.snowflake_security_integrations.test", "security_integrations.0.describe_output.0.todo.0.value"),
+					resource.TestCheckResourceAttr("data.snowflake_security_integrations.test", "security_integrations.0.describe_output.0.enabled.#", "1"),
```
Comment (sfc-gh-asawicki) [inline pkg/datasources/security_integrations_acceptance_test.go:52]: why did you change this todo without adding any more checks?

File: pkg/resources/saml2_integration_acceptance_test.go [hunk 2]
```diff
@@ -78,17 +78,17 @@ func TestAcc_Saml2Integration_basic(t *testing.T) {
 				ConfigVariables: m(issuer, string(sdk.Saml2SecurityIntegrationSaml2ProviderCustom), validUrl, cert, false, false),
 				Check: resource.ComposeAggregateTestCheckFunc(
 					resource.TestCheckResourceAttr("snowflake_saml2_integration.test", "name", id.Name()),
-					resource.TestCheckResourceAttr("snowflake_saml2_integration.test", "enabled", "unknown"),
+					resource.TestCheckResourceAttr("snowflake_saml2_integration.test", "enabled", "default"),
```
Comment (sfc-gh-asawicki) [inline pkg/resources/saml2_integration_acceptance_test.go:81]: we could just use the const we have, right?

File: pkg/resources/special_values.go [hunk 3]
```diff
@@ -37,3 +37,7 @@ func booleanStringToBool(value string) (bool, error) {
 func booleanStringFieldDescription(description string) string {
 	return fmt.Sprintf(`%s Available options are: "%s" or "%s". When the value is not set in the configuration the provider will put "%s" there which means to use the Snowflake default for this value.`, description, BooleanTrue, BooleanFalse, BooleanDefault)
 }
+
+func externalChangesNotDetectedFieldDescription(description string) string {
```
Comment (sfc-gh-asawicki) [inline pkg/resources/special_values.go:41]: 👍

**Review by sfc-gh-asawicki — DISMISSED** [review]

File: pkg/datasources/security_integrations_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-jcieslak) [reply pkg/datasources/security_integrations_acceptance_test.go:52]: Yeah, my bad, but as far as I know the data source will be done at the end, so we can add them there 👀

File: pkg/resources/saml2_integration_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-jcieslak) [reply pkg/resources/saml2_integration_acceptance_test.go:81]: yeah, but there's strange conflict. We have our resource names needed for CheckDestroy under "resources" package and our main package for resources is also named "resources". Because I was unsure how do we want to resolve this, I just used plain values.

**Review by sfc-gh-jmichalak — APPROVED** [review]

File: pkg/resources/saml2_integration_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-asawicki) [reply pkg/resources/saml2_integration_acceptance_test.go:81]: They are already used with this problem solved (named `r` import)

File: pkg/datasources/security_integrations_acceptance_test.go (hunk 1 above)
Comment (sfc-gh-asawicki) [reply pkg/datasources/security_integrations_acceptance_test.go:52]: Yeah, but we keep them usually for a reason :) that's fine with this one ofc

File: pkg/resources/saml2_integration_acceptance_test.go (hunk 2 above)
Comment (sfc-gh-jcieslak) [reply pkg/resources/saml2_integration_acceptance_test.go:81]: Addressed in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2931
